core/src/main/java/org/geysermc/geyser/session/cache/RegistryCache.java

```diff
@@ -149,25 +149,31 @@ public void load(ClientboundRegistryDataPacket packet) {
      * @param reader converts the RegistryEntry NBT into a class file
      * @param <T> the class that represents these entries.
      */
-    private static <T> void register(String registry, Function<RegistryCache, JavaRegistry<T>> localCacheFunction, BiFunction<GeyserSession, RegistryEntry, T> reader) {
-        Key key = MinecraftKey.key(registry);
-        REGISTRIES.put(key, (registryCache, entries) -> {
+    private static <T> void register(String registry, Function<RegistryCache, JavaRegistry<T>> localCacheFunction, TriFunction<GeyserSession, Function<Key, Integer>, RegistryEntry, T> reader) {
+        Key registryKey = MinecraftKey.key(registry);
+        REGISTRIES.put(registryKey, (registryCache, entries) -> {
             Map<Key, NbtMap> localRegistry = null;
             JavaRegistry<T> localCache = localCacheFunction.apply(registryCache);
             // Clear each local cache every time a new registry entry is given to us
             // (e.g. proxy server switches)
+
+            Map<Key, Integer> entryIdMap = new HashMap<>();
```

Comment:
Just add a brief comment about this being for readers that have to look back into the registry (enchants)